### PR TITLE
Add span debug logging to _on_span_finish

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
This is done in `tracer.write()` but we should also do it here since `tracer.write()` is going to be deprecated and removed soon.

### CODE REVIEW

**Review by Yun-Kim — APPROVED** [review]

**Review by brettlangdon — COMMENTED** [review]
the way the title and description are written, are you intending to add new/additional logging, or just update this existing format string?

**Comment by Kyle-Verhoog** [comment]
> are you intending to add new/additional logging, or just update this existing format string?

both

The pprinting wasn't copied over to the _on_span_finish logging and we didn't have any integration tests covering it so it was lost in #2038

**Review by Yun-Kim — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Comment by Kyle-Verhoog** [comment]
Strange - the added test fails for Python 3.7+ 🤔

**Comment by Yun-Kim** [comment]
> Strange - the added test fails for Python 3.7+ 🤔

Curious - what was the failure reason / fix? Can't figure out since it's one commit 😆

**Comment by Kyle-Verhoog** [comment]
I suspect it has to do with setting the log level 🤔

**Review by brettlangdon — APPROVED** [review]

File: tests/integration/test_debug.py [hunk 1]
```diff
@@ -359,3 +359,15 @@ def test_error_output_ddtracerun():
     stderr = p.stderr.read()
     assert b"DATADOG TRACER CONFIGURATION" not in stderr
     assert b"DATADOG TRACER DIAGNOSTIC - Agent not reachable" not in stderr
+
+
+def test_debug_span_log():
```
Comment (P403n1x87) [inline tests/integration/test_debug.py:364]: Apart from `env`, this is similar to `tests.utils.call_program`. Perhaps adapt and use that?

Comment (Kyle-Verhoog) [reply tests/integration/test_debug.py:364]: hmm I just followed the convention in the file 🤷

Comment (P403n1x87) [reply tests/integration/test_debug.py:364]: Ah I see! Sorry, forget it then!

**Review by P403n1x87 — APPROVED** [review]

**Pull request merged**